### PR TITLE
completed task 4 +  /api now return the whole endpoints.json file ins…

### PR DESCRIPTION
**Pull request by Sotogan**
…tead of only the description

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -41,13 +41,56 @@ describe('GET /api', () => {
...
+                {
+                  title: "Eight pug gifs that remind me of mitch",
+                  topic: "mitch",
+                  author: "icellusedkars",
+                  body: "some gifs",
+                  article_img_url:
+                    "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+                })
+           )
+
```
Comment (Harrison365) [inline __tests__/app.test.js:71]: We need to check that the article id that is returned is 3.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -41,13 +41,56 @@ describe('GET /api', () => {
...
+    test('should respon with an error if given a snack_id of a  NON-valid type that does not exist in our database', () => {
+        return request(app)
+        .get('/api/articles/forklift')
+        .expect(400)
+        .then((response)=>{
+
+            const error=response.body
+            expect(error.msg).toBe('bad request')
+        })        
+    });
```
Comment (Harrison365) [inline __tests__/app.test.js:94]: Nice sad path tests :)

File: models/topics-model.js [hunk 3]
```diff
@@ -14,18 +15,24 @@ exports.selectTopics=()=>{
...
+  exports.selectArticleById=(article_id) => {
   
-      
-   return wantedInfo
+   return  db.query("SELECT * FROM articles WHERE article_id=$1;",[article_id])
+   .then((article)=>{
+    
+    if (article.rows.length===0){
+        return Promise.reject({status:404, msg:'id not found'})
+    }
```
Comment (Harrison365) [inline models/topics-model.js:31]: Nice 👍

File: __tests__/app.test.js (hunk 1 above)
Comment (Sotogan) [reply __tests__/app.test.js:71]: fixed :)

**Pull request merged**